### PR TITLE
fixes registration hook issue

### PR DESCRIPTION
**Pull request by pradeepdewda-okta**
## Description:

1. it caused post to send old form data instead of new, in case of registration errors

## PR Checklist

- [ ] Have you verified the basic functionality for this change?

- [ ] Added unit tests?

- [ ] Added e2e tests

- [ ] Did you follow our [Security Best Practices](https://oktawiki.atlassian.net/wiki/display/eng/Security+Best+practices)?

### Reviewers:
@sherizada-okta 

### Issue:

- [OKTA-388105](https://oktainc.atlassian.net/browse/OKTA-388105)

### CODE REVIEW

**Review by sherizada-okta — COMMENTED** [review]
Great, thanks. There is a gap in automated testing since this fell through the cracks, either selenium and/or testcafe. The testcafe file is IdentifyRegistration_spec.js. Can you add a test for the issue that came up?

File: src/v2/view-builder/views/EnrollProfileView.js [hunk 1]
```diff
@@ -14,7 +14,7 @@ const Body = BaseForm.extend({
     // SIW customization hook for registration
     this.settings.preRegistrationSubmit(this.model.toJSON(),
```
Comment (haishengwu-okta) [inline src/v2/view-builder/views/EnrollProfileView.js:15]: In V1, we sends `this.model.attributes` to this callback
https://github.com/okta/okta-signin-widget/blob/master/src/RegistrationController.js#L149-L150

Comment (sherizada-okta) [reply src/v2/view-builder/views/EnrollProfileView.js:15]: > In V1, we sends `this.model.attributes` to this callback
> https://github.com/okta/okta-signin-widget/blob/master/src/RegistrationController.js#L149-L150

@haishengwu-okta we had this conversation https://github.com/okta/okta-signin-widget/pull/1747#discussion_r594507570

File: test/testcafe/spec/IdentifyRegistration_spec.js [hunk 2]
```diff
@@ -230,7 +247,46 @@ test.requestHooks(mock)('should show register page directly and be able to creat
...
+  await registrationPage.clickRegisterButton();
+  req = logger.requests[1].request;
+  reqBody = JSON.parse(req.body);
+
+  await t.expect(reqBody).eql({
+    stateHandle: '01OCl7uyAUC4CUqHsObI9bvFiq01cRFgbnpJQ1bz82',
+    userProfile: {
+      firstName: 'abc',
+      lastName: 'xyz',
+      email: 'newFoo@ex.com'
```
Comment (sherizada-okta) [inline test/testcafe/spec/IdentifyRegistration_spec.js:284]: sweet

**Review by sherizada-okta — APPROVED** [review]

**Comment by sherizada-okta** [comment]
You can change the base of this PR without creating a new PR. 

1. git rebase --onto origin/5.6 master pdewda-OKTA-388105-parseSchema-fix
2. On github, where the PR title is, click "Edit" then click "master" and select the right branch.

**Comment by pradeepdewda-okta** [comment]
> You can change the base of this PR without creating a new PR.
> 
> 1. git rebase --onto origin/5.6 master pdewda-OKTA-388105-parseSchema-fix
> 2. On github, where the PR title is, click "Edit" then click "master" and select the right branch.

Lovely.. Now i know why step 2 didn't work :)

**Pull request closed without merging**